packages/ui/src/electron/components/home.tsx

```diff
@@ -9,22 +9,24 @@ interface State {
 }
 
 class Home extends Component<Props, State> {
-  _isMounted = false;
+  private mutableIsMounted: boolean = false;
```

Comment:
Using the `isMounted` attribute like this is a pretty common pattern in React, but I had to significantly loosen the tslint restrictions on mutability to use it. A slightly cleaner solution would be to make`synchronizeWithCourier` and `pollConnectionStatus` return an interface with an abort() or cancel() which ensures we don't get any callbacks once this element has been "unmounted" (permanently removed from the DOM).

Reply:
Either sounds good to me. Alternatively, you could do something similar to the abort/cancel methods, without implementing them in the result of those functions, since `break`ing the iterables should close the underlying connection and stop yielding new values anyway.

By `break`ing I don't necessarily mean having a literal `break` statement. For example, you could wrap the result of `synchronizeWithCourier()`/`pollConnectionStatus()` in an [abortable-iterator](https://www.npmjs.com/package/abortable-iterator), roughly like this:

```typescript
import AbortController from 'abort-controller';
import { source as makeSourceAbortable } from 'abortable-iterator';

const abortController = new AbortController();

const abortableConnectionStatusStream = makeSourceAbortable(
  pollConnectionStatus(),
  abortController.signal,
  { returnOnAbort: true },
);
```

Then call `abortController.abort()` when the element is unmounted.

I've never used the pattern above with React/observables, so it may not be an ideal solution!

Reply:
👍  That's more or less what I was thinking too. Will do.